1. Data Processing

In [ ]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report
from sklearn.preprocessing import StandardScaler

df = pd.read_csv("/content/BankCustomerData.csv")
print(df.head())

print(df.isnull().sum())

df_dummies = pd.get_dummies(df, drop_first=True)
df_dummies['minimum_balance'] = (df['balance']>
                                 2000).astype(int)


FileNotFoundError: [Errno 2] No such file or directory: '/content/BankCustomerData.csv'

2. Feature Selection

In [ ]:
x = df_dummies.drop(['age', 'balance', 'day', 'duration', 'campaign', 'pdays', 'previous', 'minimum_balance'],axis=1)
y = df_dummies['minimum_balance']

3. data Splitting

In [ ]:
x_train, x_test, y_train, y_test = train_test_split(x,y,test_size=0.2, random_state=80)

4. Model Training

In [ ]:
scaler=StandardScaler()
x_train_scaled = scaler.fit_transform(x_train)
x_test_scaled = scaler.fit_transform(x_test)

5. Model Evaluation

In [ ]:
model = LogisticRegression()
model.fit(x_train_scaled, y_train)

y_pred = model.predict(x_test_scaled)

accuracy = accuracy_score(y_test, y_pred)
conf_matrix = confusion_matrix(y_test, y_pred)
class_report = classification_report(y_test, y_pred)

print(f"Accuracy: {accuracy}")
print("Confusion Matrix: ")
print(conf_matrix)
print("Classification Report: ")
print(class_report)

Accuracy: 0.8174249530956847
Confusion Matrix: 
[[6958   12]
 [1545   13]]
Classification Report: 
              precision    recall  f1-score   support

           0       0.82      1.00      0.90      6970
           1       0.52      0.01      0.02      1558

    accuracy                           0.82      8528
   macro avg       0.67      0.50      0.46      8528
weighted avg       0.76      0.82      0.74      8528



5. Conclusion

 The model's performance display the balance as the independent variable, wherein it happened to have an accuracy of 82% as a result. Thus, the bank's marketing strategies helps the customers to be informed enough to be able to meet the requirements to be an eligable customer before conducting a term deposit.